### DISCUSSION TITLE
How to determine a package's source repository via API or Client SDK?

### DISCUSSION THREAD

**Discussion by SeanKilleen** [NuGet/Home#11327] [Q&A] [answered]
Hi all,

I am attempting to locate the source repository URL of a NuGet package via the API or client SDK.

I'd assumed it would be returned with the `PackageMetadataResource`, but it looks like that only holds a project URL and not a source repo URL.

I've looked around in the client SDK and API and maybe I'm just missing it, but could someone point me to how to find my way to the highlighted link below programmatically?

> ![image](https://user-images.githubusercontent.com/2148318/138447072-7323fdea-b8b2-47db-863b-415811a752a0.png)

Thanks!

**Comment by SeanKilleen** [comment]
Cross-posted on SO in clase anyone prefers to answer over there: https://stackoverflow.com/questions/69676458/how-can-i-determine-a-nuget-package-source-repository-url-from-the-api-or-client

**Comment by michael-hawker** [comment]
Yeah, I was just wondering the difference between "Project Website" and "Source Repository" on nuget.org as well and where that information is coming from. As the [NuGet Server API for package metadata](https://learn.microsoft.com/nuget/api/registration-base-url-resource) only lists `projectUrl` which seems a bit ambiguous or maybe towards the former over the latter.

Any insight or clarity on the distinction of these two properties and how to retrieve/distinguish them (if possible in the API) would be great to know. Thanks!

FYI @nkolev92 is this something you can shed some light on for us? I'm trying to look up package dependencies and query NuGet to see if we can get a link to the GitHub/source for further inquiry/code-analysis; so, distinguishing this data is important for our scenario. Thanks! 🦙❤️

**Reply by michael-hawker** [reply]
Was just thinking about this more from another angle, does nuget.org just read directly from the nuspec/manifest and parse/grab the repository info from there? https://learn.microsoft.com/nuget/api/package-base-address-resource#download-package-manifest-nuspec

**Reply by nkolev92** [reply, marked as answer]
There are a bunch of concepts that are related, but not always equivalent. Nearly all of the metadata nuget.org exposes is read from the nuspec, save for something like owners which is a construct that nuget.org has.

- `project website` would be something like: https://www.newtonsoft.com/json. Equivalent to `projectUrl` from the nuspec.  This is available in the metadata resource.
- `source repository` refers to the `repository` metadata in the nuspec, which usually is supposed to refer to the code. https://learn.microsoft.com/en-us/nuget/reference/nuspec#repository. This is not available in the metadata resource: https://github.com/NuGet/NuGetGallery/issues/6262. I think you'd need to use the content resource to download either the nupkg or the nuspec.

**Reply by michael-hawker** [reply]
Thanks @nkolev92 that's great info! Mind marking that as the answer for this discussion? (Or is that feature not enabled for your Q&A section?)
